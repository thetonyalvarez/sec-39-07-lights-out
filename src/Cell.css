.Cell {
  height: 100px;
  width: 100px;
  cursor: pointer;
  /* background-color: darkgray; */
  background-image: url('./src/assets/bulb-off.png');
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
}

.Cell-lit {
  /* background-color: white; */
  background-image: url('./src/assets/bulb-on.png');
}