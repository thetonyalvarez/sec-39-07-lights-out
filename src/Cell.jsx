import React from "react";
import TableCell from '@mui/material/TableCell'
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 * 
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell({ flipCellsAroundMe, isLit }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  return <TableCell
          className={classes}
          onClick={flipCellsAroundMe}
          data-testid="cell"
          sx={{ borderBottom: 0 }}
        />;
}

export default Cell;
