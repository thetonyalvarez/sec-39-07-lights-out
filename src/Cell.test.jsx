import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import Cell from "./Cell"

it("renders without crashing", function() {
  render(<Cell/>)
});

it("matches snapshot", function() {
  const {asFragment} = render(<Cell />);
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Cell' class", function() {
  const { queryByTestId } = render(<Cell/>)

  const cell = queryByTestId('cell')
  expect(cell).toHaveClass('Cell')
})