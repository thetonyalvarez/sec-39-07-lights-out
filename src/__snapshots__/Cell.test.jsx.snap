// Vitest Snapshot v1

exports[`matches snapshot 1`] = `<DocumentFragment />`;
